src/components/App.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import Display from './Display';
+import Panel from './Panel';
+import calculate from '../logic/calculate';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+  handleClick(btn) {
+    this.setState(calculate(this.state, btn));
```

Comment:
There's a better way to do this since we are using the state to generate the new state:

```javascript
handleclick = (btn) => this.setState(prevState => calculate(prevState, btn));
```
You can find out more about the functional format of `setState` [here](https://www.freecodecamp.org/news/functional-setstate-is-the-future-of-react-374f30401b6b/).

Reply:
In this particular case I don't see why the functional format is better. It'll only add a small unnecessary complexity to the code. But thanks for the info, tho. Very good to know about it.

Reply:
Again maybe more **practical experience** will help you understand the difference of it later on :)